pxtlib/docsrender.ts

```diff
@@ -418,13 +418,27 @@ namespace pxt.docs {
 
     export function setupRenderer(renderer: marked.Renderer) {
         renderer.image = function (href: string, title: string, text: string) {
-            let out = '<img class="ui image" src="' + href + '" alt="' + text + '"';
-            if (title) {
-                out += ' title="' + title + '"';
+
+            if (href.startsWith("https://www.youtube.com/embed/")){
+
+                // <iframe width="1358" height="773" src="https://www.youtube.com/embed/0Gl2QnHNpkA?list=RDMM8H1iMq0g31g" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
+                if (/^[a-zA-Z0-9]+(?:\?list=[a-zA-Z0-9]+)?$/.test(href.substring("https://www.youtube.com/embed/".length))){
```

Comment:
What's this regex doing? Might be worth to check it into https://github.com/microsoft/pxt/blob/master/pxtlib/commonutil.ts

Reply:
ended up getting rid of regex and am constructing url, should have used better commenting, thanks!